sql/mysqld.cc

```diff
@@ -2084,16 +2089,22 @@ static void clean_up(bool print_message)
 */
 static void wait_for_signal_thread_to_end()
 {
-  uint i;
+  uint i, n_waits= DBUG_EVALUATE("force_sighup_processing_timeout", 5, 100);
   /*
     Wait up to 10 seconds for signal thread to die. We use this mainly to
     avoid getting warnings that my_thread_end has not been called
```

Comment:
The comment might be a little outdated by now. There is a potentially long-ish operation SIGHUP handler that needs to be waited on.